File: fontend/lib/models/category.dart

class Category {
  String? id;
  String? name;
  Category({this.id, this.name});

  Category.fromJson(Map<String, dynamic> json) {
    id = json['_id'];
    name = json['name'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = Map<String, dynamic>();
    data['_id'] = id;
    data['name'] = name;
    return data;
  }
}

File: backend/src/controller/Category.js
const categoryModel = require("../models/Category");

const allCategory = async (req, res, next) => {
  try {
    const categories = await categoryModel.category.find();
    return res
      .status(200)
      .json({ code: 200, result: categories, message: "success" });
  } catch (error) {
    console.log(error);
    return res.status(500).json({ code: 500, message: "internal error" });
  }
};

const addCategory = async (req, res, next) => {
  try {
    const { name } = req.body;
    const newData = new categoryModel.category({ name });
    await newData.save();
    return res.status(201).json({ code: 201, message: "success" });
  } catch (error) {
    console.log(error);
    return res.status(500).json({ code: 500, message: "internal error" });
  }
};

module.exports = {
  allCategory,
  addCategory,
};
